lib/Dialect/SPIRV/Serialization/Serializer.cpp

```diff
@@ -1523,6 +1553,29 @@ Serializer::processOp<spirv::ExecutionModeOp>(spirv::ExecutionModeOp op) {
                                operands);
 }
 
+template <>
+LogicalResult
+Serializer::processOp<spirv::MemoryBarrierOp>(spirv::MemoryBarrierOp op) {
+  auto loc = op.getLoc();
+  auto memoryScopeIntAttr =
+      op.getAttrOfType<IntegerAttr>(spirv::attributeName<spirv::MemoryScope>());
+  auto memorySemanticsIntAttr = op.getAttrOfType<IntegerAttr>(
+      spirv::attributeName<spirv::MemorySemantics>());
+
+  auto memoryScopeID = prepareConstantInt(loc, memoryScopeIntAttr);
```

Comment:
Same here. Maybe the processing of the memoryScope and memorySemantics can be commoned out?

Reply:
@MaheshRavishankar thanks for mention this! I'll update it!